src/nrtk_explorer/app/transforms.py

```diff
@@ -221,8 +219,7 @@ def compute_annotations(self, ids):
         return predictions
 
     def on_current_num_elements_change(self, current_num_elements, **kwargs):
-        dataset = get_dataset(self.state.current_dataset)
-        ids = [img["id"] for img in dataset["images"]]
+        ids = [img["id"] for img in self.context.dataset.imgs.values()]
```

Comment:
```python
ids = list(self.context.dataset.images().lookup('id'))
```

Will be much more efficient in an SQL context, and effectively the same in the json context.